### ISSUE TITLE
Python API won't export universes specified only in lattice "outer"

### ISSUE THREAD

**Issue by smharper** [mit-crpg/openmc#413]
If a universe is only specified in the `outer` element of a lattice, the Python API won't export that universe to XML.  For example, the API example for hex lattices contains

```python
lattice.universes = \
    [ [ [univ2] + [univ3]*11, [univ2] + [univ3]*5, [univ3] ],
      [ [univ2] + [univ1]*11, [univ2] + [univ1]*5, [univ1] ],
      [ [univ2] + [univ3]*11, [univ2] + [univ3]*5, [univ3] ] ]
lattice.outer = univ2
```

If you replace the `univ2` elements with `univ3` in lattice.universes:

```python
lattice.universes = \
    [ [ [univ3] + [univ3]*11, [univ3] + [univ3]*5, [univ3] ],
      [ [univ3] + [univ1]*11, [univ3] + [univ1]*5, [univ1] ],
      [ [univ3] + [univ3]*11, [univ3] + [univ3]*5, [univ3] ] ]
lattice.outer = univ2
```

then build the XML and run OpenMC, you'll get the error "ERROR: Invalid universe number 3 specified on lattice 5" because `univ2` aka "universe number 3" was never exported.

**Comment by wbinventor** [comment]
I think the reason for this may be that the ``RectLattice`` and ``HexLattice`` ``export_to_xml()`` routines never call ``export_to_xml()`` on the ``outer`` universe as they do for all of the universes in each lattice cell (see [here](https://github.com/mit-crpg/openmc/blob/develop/openmc/universe.py#L1224)). If this is indeed the issue, adding the following code below lines 937 and 1199 should do the trick:

```python
universe.create_xml_subelement(xml_element)
```

@smharper since you seem to have some input files which challenge this capability, I'll let you try out this fix.

In addition, since you bring this up I've realized another (related) issue with the ``outer`` attribute. The [``Lattice.get_all_universes()``](https://github.com/mit-crpg/openmc/blob/develop/openmc/universe.py#L752) and [``Lattice.get_unique_universes()](https://github.com/mit-crpg/openmc/blob/develop/openmc/universe.py#L693) routines do not include ``outer`` in their return values. You might consider fixing this in a PR as well if my proposal above works for you.

**Comment by smharper** [comment]
Adding `self._outer.create_xml_subelement(xml_element)` where you suggested solves my issue.

I looked at `get_all_universes()` and `get_unique_universes()`, but they don't seem to be working at all right now.  At least for the hex example, the code `self._universes.ravel()` returns a Numpy array of lists, not a Numpy array of `Universe`s.  This seems weird to me.  It looks like the `dtype=Universe` in the universes setter is being completely ignored.  If I had to guess, I would say that Numpy is unhappy that we are passing it arrays of different lengths.

Why did we decide to support hex lattices again?  :stuck_out_tongue:
